docs/data/joy/components/autocomplete/autocomplete.md

```diff
@@ -0,0 +1,355 @@
+---
+product: joy-ui
+title: React Autocomplete component
+githubLabel: 'component: autocomplete'
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/
+---
+
+# Autocomplete
+
+<p class="description">The autocomplete is a normal text input enhanced by a panel of suggested options.</p>
+
+The widget is useful for setting the value of a single-line textbox in one of two types of scenarios:
+
+1. The value for the textbox must be chosen from a predefined set of allowed values, e.g., a location field must contain a valid location name: [combo box](#combo-box).
+2. The textbox may contain any arbitrary value, but it is advantageous to suggest possible values to the user, e.g., a search field may suggest similar or previous searches to save the user time: [free solo](#free-solo).
+
+It's meant to be an improved version of the "react-select" and "downshift" packages.
+
+{{"component": "modules/components/ComponentLinkHeader.js"}}
+
+## Combo box
+
+The `Autocomplete` renders an input and a popup, after triggered by a pointer or keyboard, that displays a list of items from the `options` prop. The value must be chosen from a predefined set of allowed values.
+
+{{"demo": "ComboBox.js"}}
+
+### Options structure
+
+By default, the component accepts the following structures for `options`:
+
+```ts
+interface AutocompleteOption {
+  label: string;
+}
+// or
+type AutocompleteOption = string;
+```
+
+for instance:
+
+```js
+const options = [
+  { label: 'The Godfather', id: 1 },
+  { label: 'Pulp Fiction', id: 2 },
+];
+// or
+const options = ['The Godfather', 'Pulp Fiction'];
+```
+
+However, you can use different structures by providing a `getOptionLabel` prop:
+
+```js
+const options = [
+  { title: 'Pulp Fiction', id: 2 },
+  // ...
+];
+
+<Autocomplete getOptionLabel={option => option.title}>
+```
+
+### Controlled states
+
+The component has two states that can be controlled:
+
+1. the "value" state with the `value`/`onChange` props combination. This state represents the value selected by the user, for instance when pressing <kbd class="key">Enter</kbd>.
+2. the "input value" state with the `inputValue`/`onInputChange` props combination. This state represents the value displayed in the textbox.
+
+:::warning
+⚠️ These two states are isolated, they should be controlled independently.
+:::
+
+{{"demo": "ControllableStates.js"}}
+
+## Label
+
+Put an `Autocomplete`, a `FormLabel` and a `FormHelperText` (optional) under a `FormControl` component.
+
+{{"demo": "LabelAndHelperText.js"}}
+
+## Features
```

Comment:
This is awesome! I wonder if we could have a similar feature list for all the other components, included on the playground demo? By the way, feel like this should totally be the playground introduction demo, though.

Reply:
> this should totally be the playground introduction demo

Good idea! I have moved this to the [introduction](https://deploy-preview-34315--material-ui.netlify.app/joy-ui/react-autocomplete/#introduction) section.

Reply:
Sweet, this is nice! Good job at sticky positioning this as well!